java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -297,37 +298,83 @@ private fun KotlinFileExtractor.extractBinaryExpression(
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_divexpr)
+    } else if (op == KtTokens.PERC && target.isNumericWithName("rem")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_remexpr)
+    } else if (op == KtTokens.IDENTIFIER && target.isNumericWithName("and")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_andbitexpr)
+    } else if (op == KtTokens.IDENTIFIER && target.isNumericWithName("or")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_orbitexpr)
+    } else if (op == KtTokens.IDENTIFIER && target.isNumericWithName("xor")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_xorbitexpr)
+    } else if (op == KtTokens.IDENTIFIER && target.isNumericWithName("shl")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_lshiftexpr)
+    } else if (op == KtTokens.IDENTIFIER && target.isNumericWithName("shr")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_rshiftexpr)
+    } else if (op == KtTokens.IDENTIFIER && target.isNumericWithName("ushr")) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_urshiftexpr)
+    } else if (op == KtTokens.EQEQEQ && target == null) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_eqexpr)
+    } else if (op == KtTokens.EXCLEQEQEQ && target == null) {
+        extractBinaryExpression(expression, callable, parent, tw::writeExprs_neexpr)
+    } else if (op in listOf(KtTokens.LT, KtTokens.GT, KtTokens.LTEQ, KtTokens.GTEQ)) {
+        if (target.isNumericWithName("compareTo")) {
+            when (op) {
+                KtTokens.LT -> extractBinaryExpression(expression, callable, parent, tw::writeExprs_ltexpr)
+                KtTokens.GT -> extractBinaryExpression(expression, callable, parent, tw::writeExprs_gtexpr)
+                KtTokens.LTEQ -> extractBinaryExpression(expression, callable, parent, tw::writeExprs_leexpr)
+                KtTokens.GTEQ -> extractBinaryExpression(expression, callable, parent, tw::writeExprs_geexpr)
+                else -> TODO("error")
             }
+        } else {
+            TODO("Extract lowered equivalent call, such as `a.compareTo(b) < 0`")
         }
 
-        else -> TODO()
+    } else {
+        // todo: other operators, such as .., ..<, in, !in, =, +=, -=, *=, /=, %=, ==, !=,
+        TODO("Extract as method call")
     }
+}
+
+private fun KaFunctionSymbol?.isBinaryPlus(): Boolean {
+    return this != null && (
+            this.isNumericWithName("plus") ||
+                    this.hasName("kotlin", "String", "plus") ||
+                    this.hasMatchingNames(
+                        CallableId(FqName("kotlin"), null, Name.identifier("plus")),
+                        ClassId(FqName("kotlin"), Name.identifier("String")),
+                        nullability = KaTypeNullability.NULLABLE,
```

Comment:
I think it would be worth adding a comment saying what this last case corresponds to

Reply:
I can dig that up, but this bit was just refactored. It was originally introduced in https://github.com/github/codeql/pull/17752

Reply:
Added in https://github.com/github/codeql/pull/17761/commits/53460d7ca046976359eb2617498e2860210cb321.